base/server/src/main/java/com/netscape/cmscore/dbs/Repository.java

```diff
@@ -697,8 +715,11 @@ public void checkRanges() throws EBaseException {
             logger.debug("Repository: Requesting next range");
             String nextRange = getNextRange();
             logger.debug("Repository: next range: " + nextRange);
-
-            mNextMinSerialNo = new BigInteger(nextRange, mRadix);
+            if (idGenerator == IDGenerator.LEGACY_2) {
+                mNextMinSerialNo = new BigInteger(nextRange);
+            } else {
+                mNextMinSerialNo = new BigInteger(nextRange, mRadix);
```

Comment:
Could you also add a note here saying that the `nextRange` is always decimal and the code that parses the value as hexadecimal for SSNv1 is incorrect?

Reply:
With the new code it is decimal. In SSNv1 not sure if the error is here to read as hex or when it is read as decimal (as for the PR #4840). I'll add a comment to explain.